AutoRest/Generators/Ruby/Azure.Ruby/TemplateModels/PageTemplateModel.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Rest.Generator.ClientModel;
+using Microsoft.Rest.Generator.Ruby;
+
+namespace Microsoft.Rest.Generator.Azure.Ruby
+{
+    public class PageTemplateModel : AzureModelTemplateModel
+    {
+        public PageTemplateModel(CompositeType source, ISet<CompositeType> allTypes, string nextLinkName, string itemName)
+            : base(source, allTypes)
+        {
+            this.NextLinkName = nextLinkName;
+            this.ItemName = itemName;
+            this.NextLinkParameter = BuildNextLinkParameter();
+        }
+
+        public string NextLinkName { get; private set; }
+
+        public Parameter NextLinkParameter { get; private set; }
+
+        public string ItemName { get; private set; }
+
+        public CompositeType ItemType
+        {
+            get
+            {
+                if (Properties == null)
+                {
+                    return null;
+                }
+                Property property = Properties.FirstOrDefault(p => p.Type is SequenceType);
+                if (property != null)
+                {
+                    return ((SequenceType)property.Type).ElementType as CompositeType;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        public Parameter BuildNextLinkParameter()
+        {
+            RubyCodeNamer rubyCodeNamer = new RubyCodeNamer();
+            Parameter param = new Parameter
```

Comment:
I am not sure but i think we should not new RubyCodeNamer. Is there any chance we can get the normalized NextLinkName from the AzureRubyCodeGenerator class itself? what do you think about that?